pkg/api/validation/validation.go

```diff
@@ -4474,3 +4475,27 @@ func validateStorageNodeAffinityAnnotation(annotations map[string]string, fldPat
 	}
 	return policySpecified, allErrs
 }
+
+func validateSecurityContext(sc *api.SecurityContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if sc == nil {
+		return allErrs
+	}
+
+	if sc.AllowPrivilegeEscalation != nil && !*sc.AllowPrivilegeEscalation {
+		if sc.Privileged != nil && *sc.Privileged {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath(""), "Cannot set `allowPrivilegeEscalation` to false and `privileged` to true"))
+		}
+
+		if sc.Capabilities != nil {
+			for _, cap := range sc.Capabilities.Add {
+				if string(cap) == "CAP_SYS_ADMIN" {
+					allErrs = append(allErrs, field.Forbidden(field.NewPath(""), "Cannot set `allowPrivilegeEscalation` to false and `capAdd` CAP_SYS_ADMIN"))
```

Comment:
It shouldn't be `field.NewPath("")`. Parent field should be passed to this function and here we should append its child element.